### PR TITLE
doc: hardware: emulator: add gpio_emul documentation

### PR DESCRIPTION
**Pull request by cfriedt**
Previously, the only documentation that existed for gpio_emul was in the form of source code.

Create some additional documentation in the form of .rst to help clarify usage. Any additional instruction for how to use the GPIO Emulator driver can be added here.

### CODE REVIEW

**Comment by cfriedt** [comment]
Ready-ish I think - happy to make modifications, as needed.

File: doc/hardware/emulator/gpio_emulator.rst
Comment (aaronemassey) [inline doc/hardware/emulator/gpio_emulator.rst:17]: I am personally a fan of documentation with a conversational tone :), but I defer to other reviewers.

File: doc/hardware/emulator/gpio_emulator.rst
Comment (aaronemassey) [inline doc/hardware/emulator/gpio_emulator.rst:33]: You reminded me I need to go make a gh issue about needing support for tristate gpios.

**Review by aaronemassey — APPROVED** [review]

File: doc/hardware/emulator/gpio_emulator.rst
Comment (kartben) [reply doc/hardware/emulator/gpio_emulator.rst:17]: Are you, or are you not? Just asking for clarification in case you forgot a word -- I have to admit that the tone certainly had me do a double take yesterday, even before you made this comment, and I would lean on the not a fan side myself. That being said, I will also defer to others :) And in any case I will try to provide actionable feedback. Thanks for the work on this doc page, Chris!

**Review by aescolar — CHANGES_REQUESTED** [review]
Thanks for documenting it,
But it does not seem ready yet.
Rendered version can be found here:
https://builds.zephyrproject.io/zephyr/pr/67410/docs/hardware/emulator/gpio_emulator.html

File: doc/hardware/emulator/gpio_emulator.rst
Comment (kartben) [reply doc/hardware/emulator/gpio_emulator.rst:18]: +1, this 100% looked like a typo to me, tbh

File: doc/hardware/emulator/gpio_emulator.rst
Comment (aescolar) [inline doc/hardware/emulator/gpio_emulator.rst:55]: Please replace zephyr,gpio-emul with a link to the binding doc page
(:dtcompatible:`zephyr,gpio-emul`)

File: doc/hardware/emulator/gpio_emulator.rst
Comment (cfriedt) [reply doc/hardware/emulator/gpio_emulator.rst:18]: Lol, sure. I guess it's more of a native English thing / colloquialism

File: doc/hardware/emulator/gpio_emulator.rst
Comment (cfriedt) [reply doc/hardware/emulator/gpio_emulator.rst:13]: It was used that way when it was added and that's how I was instructed to design it by the previous gpio maintainer.

We also do not yet have a replacement GPIO maintainer.

In any case, I will remove it.

File: doc/hardware/emulator/gpio_emulator.rst
Comment (cfriedt) [reply doc/hardware/emulator/gpio_emulator.rst:17]: For some reason, I misread "conversational" as "controversial" initially 😅 I guess both could be true.

I will make some objective corrections first, and then will look at subjective corrections.

It might take a bit of time

File: doc/hardware/emulator/gpio_emulator.rst
Comment (alevkoy) [inline doc/hardware/emulator/gpio_emulator.rst:31]: Related to the above discussion of formality in documentation, "Use the emulated GPIO driver" is fairly informal, while "one would use" is fairly formal. Without expressing a strong opinion on which one I think you should choose, this sentence and the document should be internally consistent: "Use the emulated GPIO driver ... as you would use ..." or "One uses the emulated GPIO driver ... as one would use ..."

The same idea also applies to other constructions below like "may be specified" ("you may specify") and "it may be necessary" ("you may need").

File: doc/hardware/emulator/gpio_emulator.rst
Comment (alevkoy) [inline doc/hardware/emulator/gpio_emulator.rst:87]: Maybe a snippet here that gets the interesting information out of `port` and `pins` and calls some made-up function with those objects as arguments?

**Review by alevkoy — APPROVED** [review]

**Pull request closed without merging**
